stdlib/private/SwiftReflectionTest/SwiftReflectionTest.swift

```diff
@@ -154,9 +154,13 @@ extension Section {
 internal func getReflectionInfoForImage(atIndex i: UInt32) -> ReflectionInfo? {
 #if INTERNAL_CHECKS_ENABLED
   return _getMetadataSection(UInt(i)).map { rawPointer in
-    let name = _getMetadataSectionName(rawPointer)
+    let cName = _copyMetadataSectionName(rawPointer)
+    defer {
+      cName.deallocate()
```

Comment:
This mixes `malloc` (by way of `strdup`) with `deallocate`, which I think is supposed to be a no-no.

Reply:
Dang, was hoping I'd get away with it.

`free()` isn't available consistently on the Swift side. Would you rather I exposed a `_freeMetadataSectionName()` entry point to match `_copyMetadataSectionName()`? Or some other approach.

Reply:
A paired free function sounds good, if `free` can't be used directly.